File: 2432. The Employee That Worked on the Longest Task.cpp
class Solution {
public:
    int hardestWorker(int n, vector<vector<int>>& logs) {
        int maxDuration = 0; 
        int maxWorker = logs[0][0];
        int previousLeaveTime = 0;
        for(int i = 0; i < logs.size(); i++){
            int workerID = logs[i][0];
            int leaveTime = logs[i][1];
            int duration = leaveTime - previousLeaveTime;
            if(duration > maxDuration){
                maxWorker = workerID;
                maxDuration = duration;
            }
            else if(duration == maxDuration && workerID < maxWorker){
                maxWorker = workerID;
            }
            previousLeaveTime = leaveTime;
        }
        return maxWorker;
    };
    
};

File: 2432. The Employee That Worked on the Longest Task.js
/**
 * @param {number} n
 * @param {number[][]} logs
 * @return {number}
 */
var hardestWorker = function(n, logs) {
    let previousLeaveTime = 0;
    let maxDuration = 0;
    let maxWorker = logs[0][0];

    logs.forEach((task) => {
        let id = task[0];
        let leaveTime = task[1];
        let duration = leaveTime - previousLeaveTime;

        if (duration > maxDuration) {
            maxDuration = duration;
            maxWorker = id;
        } else if (duration === maxDuration && id < maxWorker) {
            maxWorker = id;
        }

        previousLeaveTime = leaveTime;
    });

    return maxWorker;
};
